js/btcmarkets.js

```diff
@@ -487,16 +589,26 @@ module.exports = class btcmarkets extends Exchange {
         if (api === 'private') {
             this.checkRequiredCredentials ();
             let nonce = this.nonce ().toString ();
-            // eslint-disable-next-line quotes
-            let auth = uri + "\n" + nonce + "\n";
+            let auth = undefined;
             headers = {
-                'Content-Type': 'application/json',
                 'apikey': this.apiKey,
                 'timestamp': nonce,
             };
-            if (method === 'POST') {
+            if (method === 'post') {
+                // eslint-disable-next-line quotes
+                headers['Content-Type'] = 'application/json';
+                auth = uri + '\n' + nonce + '\n';
                 body = this.json (params);
                 auth += body;
+            } else {
+                let queryStringParams = this.omit (params, this.extractParams (path));
+                let queryString = '';
+                if (Object.keys (queryStringParams).length) {
+                    let query = this.ordered (queryStringParams);
+                    queryString = this.urlencode (query);
+                    url += '?' + queryString;
+                }
+                auth = uri + '\n' + (queryString ? queryString + '\n' : '') + nonce + '\n';
```

Comment:
This is not transpilable, unfortunately, we need to simplify this line as described here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#using-ternary-conditionals

Reply:
Also, `'\n'` must be enclosed in double quotes (`"\n"`) otherwise it breaks PHP. Will add that to the guidelines as well.